### ISSUE TITLE
Proxy not being set or used after update to Docker Desktop 2.2

### ISSUE THREAD

**Issue by burtonr** [docker/for-win#5548]
- [X] I have tried with the latest version of my channel (Stable or Edge)
  - [X] I have uploaded Diagnostics
  - Diagnostics ID: 074CE892-1370-4DD0-9F2D-0CDA570237E5/20200122175706

### Expected behavior
Pull images from Docker Hub from behind a corporate proxy after setting the "Manual proxy configuration" values in the UI

### Actual behavior
Receive EOF errors when attempting to pull, or login
```sh
❯ docker pull busybox
Using default tag: latest
Error response from daemon: Get https://registry-1.docker.io/v2/: EOF

❯ docker login
Authenticating with existing credentials...
Login did not succeed, error: Error response from daemon: Get https://registry-1.docker.io/v2/: EOF                                                                                                                                                                            
```

### Information

  - Windows Version: *Windows 10 Enterprise 1809 (OS Build: 17763.914)*
  - Docker Desktop Version: *2.2.0.0 (42247)*
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM: *No*

This only just started today after everyone has updated to the new Docker Desktop 2.2. The team I work with are all excitable for Docker updates (myself included) and we've all updated. That means unfortunately, that means I can't check to be certain it is a change in Docker Desktop 2.2 and not our corporate admins having impeccable timing as always. 

Looking through the logs, I see it accessing the `http_proxy.json` file. However, when I open that file, it only contains empty curly braces `{ }`. This is true when changing the "Manual proxy configuration" on and off. The file remains unchanged.

I've also set the environment variables (in Powershell) and executed the `docker pull busybox` with the same result, and similar logs. Not sure if Windows respects those environment variables, but it is something I tried.  

I'll also mention that making changes to the proxy in the UI, then clicking the "Apply & Restart" button quickly flashes that Docker had restarted. All in less than a second... Not sure if that's a feature, or a bug ;) What I mean to say is that I don't know if it's actually making the changes and restarting that fast, or if it's a bug in that nothing is being applied and it is not restarting

### Steps to reproduce the behavior

  1. Connect to a network behind a proxy
  2. Attempt to pull an image (`busybox` in my example)
  3. Observe the `EOF` error (or similar)

**Comment by burtonr** [comment]
> tl;dr; - Run "Reset to factory defaults" and add username and password to the proxy URI fixed this issue for me and my team.

I will close this as the issue is resolved for me and my team now. I can't say for certain that the issue was not with the Docker Desktop upgrade, but there are definitely other factors in play in my situation.

I ended up having to run the "Reset to factory defaults" in order for the "Apply & Restart" command to function properly.
I was then able to add my proxy information in the UI and was then able to pull the `busybox` image
However, some image pull commands would return with an `authorizationRequired` error. After some time, I found that I needed to add my username and password to the proxy URI in the Docker UI: `http://burtonr:superSecret@proxy.company.com`

Again, I'm not sure that the Docker upgrade caused a problem that required me to reset everything, although it was working beforehand. Also, not sure the Docker upgrade changed anything with the proxy, but I did not need a username/password before this update.

**Comment by ulrich** [comment]
Thank you for your feedback, we will investigate for the trouble.

**Comment by mikeparker** [comment]
> I'll also mention that making changes to the proxy in the UI, then clicking the "Apply & Restart" button quickly flashes that Docker had restarted. All in less than a second... Not sure if that's a feature, or a bug ;) What I mean to say is that I don't know if it's actually making the changes and restarting that fast, or if it's a bug in that nothing is being applied and it is not restarting

This sounds like the bug. Your logs may have a clue if the backend returned an error instead of restarting. Its the restart process that saves the proxy settings into your http_proxy.json file, so if thats not being updated then clearly the restart process isnt working (as you also saw with the instant flicker).

**Comment by mikeparker** [comment]
@burtonr it looks like your diagnostic didn't upload correctly. Is there any chance you still have the log file or diagnostic bundle lying around?

**Comment by burtonr** [comment]
@mikeparker apologies for not seeing this earlier. I will look for it first thing in the morning and share it with you if I'm able to find it. Thanks for looking into this!

**Comment by burtonr** [comment]
@mikeparker Hopefully these help you out:
[20200122164622.zip](https://github.com/docker/for-win/files/4130354/20200122164622.zip)
[20200122175706.zip](https://github.com/docker/for-win/files/4130357/20200122175706.zip)

I see 3 `.zip` files in that log directory. One of them was empty and had an error. That must have been the one I attempted to upload. Windows just tells me "it's empty".
